flink-connector/src/main/java/com/datastrato/gravitino/flink/connector/catalog/BaseCatalog.java

```diff
@@ -47,35 +67,81 @@ public void close() throws CatalogException {}
 
   @Override
   public List<String> listDatabases() throws CatalogException {
-    throw new UnsupportedOperationException();
+    return Arrays.stream(catalog().asSchemas().listSchemas())
+        .map(NameIdentifier::name)
+        .collect(Collectors.toList());
   }
 
   @Override
-  public CatalogDatabase getDatabase(String s) throws DatabaseNotExistException, CatalogException {
-    throw new UnsupportedOperationException();
+  public CatalogDatabase getDatabase(String databaseName)
+      throws DatabaseNotExistException, CatalogException {
+    try {
+      Schema schema = catalog().asSchemas().loadSchema(databaseName);
+      return new CatalogDatabaseImpl(schema.properties(), schema.comment());
+    } catch (NoSuchSchemaException e) {
+      throw new DatabaseNotExistException(getName(), databaseName);
+    }
   }
 
   @Override
-  public boolean databaseExists(String s) throws CatalogException {
-    throw new UnsupportedOperationException();
+  public boolean databaseExists(String databaseName) throws CatalogException {
+    return catalog().asSchemas().schemaExists(databaseName);
   }
 
   @Override
-  public void createDatabase(String s, CatalogDatabase catalogDatabase, boolean b)
+  public void createDatabase(
+      String databaseName, CatalogDatabase catalogDatabase, boolean ignoreIfExists)
       throws DatabaseAlreadyExistException, CatalogException {
-    throw new UnsupportedOperationException();
-  }
-
-  @Override
-  public void dropDatabase(String s, boolean b, boolean b1)
+    try {
+      catalog()
+          .asSchemas()
+          .createSchema(
+              databaseName, catalogDatabase.getComment(), catalogDatabase.getProperties());
```

Comment:
do you need to handle some schema property convert? such as schema property `location`

Reply:
@mchades Why do we need to convert? The properties of schema in Flink doesn't have no restrictions. Can you explain more details. Thanks.

Reply:
If users need to specify a location when creating a Hive database through Flink, is it also done by letting users specify it in the schema properties?

Reply:
see comment: https://github.com/datastrato/gravitino/pull/3582#discussion_r1624305532